chain/chain/src/approval_verification.rs

```diff
@@ -0,0 +1,97 @@
+use near_chain_primitives::error::Error;
+use near_crypto::Signature;
+use near_primitives::{
+    block_header::{Approval, ApprovalInner},
+    epoch_info::EpochInfo,
+    errors::EpochError,
+    hash::CryptoHash,
+    types::{AccountId, ApprovalStake, Balance, BlockHeight},
+};
+use std::{collections::HashSet, sync::Arc};
+
+/// Gets block approvers for epoch
+fn get_heuristic_block_approvers_ordered(
```

Comment:
nit: I usually try to keep private helper functions after the `pub` once, this way it is easier to see the public api exposed in this mod